### PR TITLE
ch1section0 edits

### PR DESCRIPTION
**Pull request by dankamongmen**
[sending edits by the section for manageability]

Changed up rather more content than I would like here due to style, but it's the beginning of the book; it ought punch a bit harder imho. Feel free to reject any changes you don't like. I did think a few things needed fixing: microarchitecture improvements did a lot, if not as much as frequency boosts. Moore's law has *not* continued according to its original track, but slowed. From wikipedia (ugh):

  Microprocessor architects report that semiconductor advancement has
  slowed industry-wide since around 2010, below the pace predicted by
  Moore's law.[17] Brian Krzanich, the former CEO of Intel, announced,
  "Our cadence today is closer to two and a half years than two."[103]
  Intel stated in 2015 that improvements in MOSFET devices have slowed,
  starting at the 22 nm feature width around 2012, and continuing at 14
  nm.[104] Pat Gelsinger, Intel CEO, stated at the end of 2023 that "we're
  no longer in the golden era of Moore's Law, it's much, much harder now,
  so we're probably doubling effectively closer to every three years now,
  so we've definitely seen a slowing."

I mean, we've definitely seen slowing there; I don't see how you can argue otherwise. Compiler improvements haven't achieved much, unfortunately.

I would maybe mention the Sprangle/Carmean 2002 paper "increasing processor performance by implementing deeper pipelines."

Maybe mention Cerebras for transistor count fun.

There weren't any grammatical issues in this section iirc, so seriously, you can dump all the wording changes if you'd rather not admit them. I'll try to do less of this from here on out.

### CODE REVIEW

File: chapters/1-Introduction/1-0 Introduction.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
 # Introduction {#sec:chapter1}
 
-They say, "Performance is king". It was true a decade ago, and it certainly is now. According to [@Domo2017], in 2017, the world has been creating 2.5 quintillion[^1] bytes of data every day, and as predicted in [@Statista2024], it will reach 400 quintillion bytes per day in 2024. In our increasingly data-centric world, the growth of information exchange fuels the need for both faster software and faster hardware.
```
Comment (dankamongmen) [inline chapters/1-Introduction/1-0 Introduction.md:3]: who says this? i've heard "cash is king", but i've never heard "performance is king" as a generally-stated truism. it seems more your assertion.

File: chapters/1-Introduction/1-0 Introduction.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+Transistor counts continue to increase in modern processors. Apple's M1 shipped in 2020 with 16 billion transistors. M2 made use of 20, M3 included 25, and 2024's M4 employs 28 billion transistors, a fifty percent increase over four years. The growth in transistor count enables manufacturers to add more cores to a processor. Intel's Sierra Forest dies are expected to boast a formidable 288 cores each. This is very impressive. Unfortunately, it doesn't always translate into better performance. Very often, application performance doesn't scale with extra CPU cores.
 
-When it's no longer the case that each hardware generation provides a significant performance boost, we must start paying more attention to how fast our code runs. When seeking ways to improve performance, developers should not rely on hardware. Instead, they should start optimizing the code of their applications.
```
Comment (dankamongmen) [inline chapters/1-Introduction/1-0 Introduction.md:15]: the wording here made it sound like the preamble to the Declaration of Independence =].

File: chapters/1-Introduction/1-0 Introduction.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 
-Software programmers have had an "easy ride" for decades, thanks to Moore’s law. It used to be the case that some software vendors preferred to wait for a new generation of hardware to speed up their software products and did not spend human resources on making improvements in their code. By looking at Figure @fig:50YearsProcessorTrend, we can see that single-threaded[^2] performance growth is slowing down. From 1990 to 2000, single-threaded performance grew by a factor of approximately 25 to 30 times based on SPECint benchmarks. The increase in CPU frequency was the key factor driving performance growth.
+Software programmers have had an "easy ride" for decades, thanks to Moore’s law. Software vendors could rely on new generations of hardware to speed up their software products, even if they did not spend human resources on making improvements. By looking at Figure @fig:50YearsProcessorTrend, we can see that single-threaded[^2] performance growth is slowing down. From 1990 to 2000, single-threaded performance on SPECint benchmarks increased by a factor of approximately 25 to 30, driven largely by higher CPU frequencies and improve microarchitecture.
```
Comment (dankamongmen) [inline chapters/1-Introduction/1-0 Introduction.md]: ack, i introduced an error here! fixing

Comment (dankamongmen) [reply chapters/1-Introduction/1-0 Introduction.md]: fixed

File: chapters/1-Introduction/1-0 Introduction.md (hunk 1 above)
Comment (dendibakh) [reply chapters/1-Introduction/1-0 Introduction.md:3]: Ok, I heard it a few times, but I agree that it's not used that frequently.

**Review by dendibakh — APPROVED** [review]
I agree with your notes re: uarch improvements and Moore's law.
However, I reverted some of your changes.
My priority would be easy-flowing text with *clear connections* even if it may sound a little clumsy.
I don't want to impress readers with my English. Rather I want them to feel good that they could understand the material and what I'm trying to say.

**Pull request merged**